x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -349,46 +349,23 @@ public void authorizeIndexAction(
             listener.onResponse(
                 new IndexAuthorizationResult(true, requestInfo.getOriginatingAuthorizationContext().getIndicesAccessControl())
             );
-        } else if (request instanceof IndicesRequest.Replaceable replaceable && replaceable.allowsRemoteIndices()) {
-            // remote indices are allowed
-            indicesAsyncSupplier.getAsync(ActionListener.wrap(resolvedIndices -> {
-                assert resolvedIndices.isEmpty() == false
-                    : "every indices request needs to have its indices set thus the resolved indices must not be empty";
-                // all wildcard expressions have been resolved and only the security plugin could have set '-*' here.
-                // '-*' matches no indices so we allow the request to go through, which will yield an empty response
-                if (resolvedIndices.isNoIndicesPlaceholder()) {
-                    // check action name
-                    listener.onResponse(authorizeIndexActionName(action, authorizationInfo, IndicesAccessControl.ALLOW_NO_INDICES));
-                } else {
-                    assert resolvedIndices.getLocal().stream().noneMatch(Regex::isSimpleMatchPattern)
-                        || replaceable.indicesOptions().expandWildcardExpressions() == false
-                        || (request instanceof AliasesRequest aliasesRequest && aliasesRequest.expandAliasesWildcards() == false)
-                        : "expanded wildcards for local indices OR the request should not expand wildcards at all";
-                    listener.onResponse(
-                        buildIndicesAccessControl(
-                            action,
-                            authorizationInfo,
-                            Sets.newHashSet(resolvedIndices.getLocal()),
-                            aliasOrIndexLookup
-                        )
-                    );
-                }
-            }, listener::onFailure));
         } else {
             try {
```

Comment:
What do you think if we also get rid of this `try-catch` and merge the two `if` blocks, by moving the `ensureRBAC(authorizationInfo)` at the start of the `authorizeIndexAction` method?

Reply:
I think we can get rid of the try-catch because the authorization process is mostly handled by the `authzInfoListener`
https://github.com/elastic/elasticsearch/blob/ecae22299b1ca177730becdd273e31986f302f9c/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java#L264-L267

which already handles try-catch internallly (because of `ActionListener.wrap`). 

I am not sure what you mean by "merge the two if blocks" though. Which two?

Reply:
Sorry for not being clear.
I think the reason for the try-catch is that ` ensureRBAC(authorizationInfo)` can theoretically throw, and that's not going to be funneled throw the listener's `onFailure`.
If we move the `ensureRBAC(authorizationInfo)` as the first statement in the method, and catch the exception there, we can get rid of all try-catch elsewhere and in this case fuse the "if" statement on line 354 with the "else" statement in line 352.

Reply:
Thanks for the clarification. Yes. I think we can and should move `ensureRBAC` to the beginning of the method. Ensuring `AuthorizationInfo` is an instance of `RBACAuthorizationInfo` should be the first thing to check which is actually the case for other methods like `authorizeClusterAction`, `authorizedRunAs` etc.